### PR TITLE
added option to use proxy server Fixing Issue #20

### PR DESCRIPTION
**Pull request by bsangwan**
Fixing issue #20 
I added option to use proxy server if needed. I added new parameter "proxy_server_name" in plugin.conf file. 
 At info log level the proxy server info will be logged at every metric push.
I also updated the Readme and setup script to collect proxy server.

### CODE REVIEW

**Comment by bsangwan** [comment]
@sebasrp, We have to add proxy support as well as a silent configurator to meet our needs. Let me know of you are okay with this PR, I will create a new PR for adding silent configurator if you are okay with it.

File: README.md [hunk 1]
```diff
@@ -29,6 +30,7 @@ region = "us-west-1"
 host = "Server1"
 whitelist_pass_through = False
 debug = False
+proxy_server_name = "http://myproxyserver.com:8080"
```
Comment (yimuniao) [inline README.md]: Why not make the port configurable?

Comment (yimuniao) [inline README.md]: I suggest also add proxy description into  collectd-cloudwatch/src/cloudwatch/config/plugin.conf, then we can easily know the configuration fields from the codes, I guess someone does not read readme carefully.

Comment (bsangwan) [inline README.md]: I will split proxy server name and server port. Currently they both are together.

Comment (bsangwan) [inline README.md]: Right, nice catch. I will add it.

**Comment by bsangwan** [comment]
@yimuniao I checked in all of your suggestions.

**Comment by yimuniao** [comment]
I realizes there is no unit test, could you add some unit tests to cover the new codes?
And are your codes under MIT license?

**Comment by bsangwan** [comment]
Yes, its MIT license. I can add the testcase, but it will take time.

**Comment by yimuniao** [comment]
Thanks. Your contributions are very helpful.

**Comment by yimuniao** [comment]
Any update?

**Comment by bsangwan** [comment]
sorry for delay, I will work on it next week.

**Comment by bsangwan** [comment]
@yimuniao, I added testcase for new properties. Let me know if I miss something.

**Comment by yimuniao** [comment]
Fantastic.
Thanks a lot.

**Pull request merged**